Cesium3DTilesSelection/src/TilesetContentManager.cpp

```diff
@@ -295,13 +294,43 @@ std::optional<RasterOverlayDetails> calcRasterOverlayDetailsInWorkerThread(
   const CesiumGeospatial::BoundingRegion* pRegion =
       getBoundingRegionFromBoundingVolume(contentBoundingVolume);
 
+  // remove any projections that are already used to generated UV
+  int32_t firstRasterOverlayTexCoord = 0;
+  if (result.rasterOverlayDetails) {
+    const auto& existingProjections =
+        result.rasterOverlayDetails->rasterOverlayProjections;
+    firstRasterOverlayTexCoord =
+        static_cast<int32_t>(existingProjections.size());
+    for (size_t i = 0; i < projections.size(); ++i) {
```

Comment:
I think this can be written more cleanly with `std::remove_if`. Something like this (totally untested):

```
projections.erase(
  std::remove_if(
    projections.begin(),
    projections.end(),
    [&existingProjections](const Projection& projection) {
      return std::find(
        existingProjections.begin(),
        existingProjections.end(),
        projection) != existingProjections.end();
    }),
  projections.end());
```

Maybe no one but me finds that cleaner? 😆

Reply:
```
/**
 * Returns whether the given container contains the given element.
 * 
 * @param c The container
 * @param t The element
 * @return Whether the container contains the element
 * @tparam <C> The `Container` type
 * @tparam <T> The element type
 */
template<typename C, typename T> 
bool contains(const C& c, const T& t) {
  return std::find(c.begin(), c.end(), t) != c.end(); 
}

/**
 * Removes from the given vector all elements that are contained
 * in the given container.
 * 
 * @param v The vector
 * @param toRemove The container containing the elements to remove
 * @tparam <C> The `Container` type
 * @tparam <T> The element type
 */
template<typename C, typename T> 
void removeAll(std::vector<T>& v, const C& toRemove) {
  auto& newEnd = std::remove_if(v.begin(), v.end(), 
    [&](const T& e) { return contains(toRemove, e); });
  v.erase(newEnd, v.end());
}
```
Call as
`removeAll(projections, existingProjections);`

Akin to [`removeAll`](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#removeAll-java.util.Collection-) / [`contains`](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#contains-java.lang.Object-). 

BTW: This has quadratic complexity. But I assume that there will be very few projections...

Reply:
Thanks @javagl! good idea, I will add this to the utility library